### DISCUSSION TITLE
Postal is not picking up my SPF and DKIM records.

### DISCUSSION THREAD

**Discussion by paulstuffins** [postalserver/postal#1559] [Help with using Postal]
Hi everyone,

I have installed Postal and am in the process of setting my first domain to be used with it.

I have set up my SPF[1] and DKIM[2] and they are validating correctly via DomainTools, however when I click the "Check my records are correct" button on the domain setup page, they both come back saying that the records do not exist.

Is there any caching of DNS records performed by Postal, and if there is how long are the results cached? Other than caching, what else could cause Postal to not validate the records?

Many Thanks
Paul

[1] https://mxtoolbox.com/SuperTool.aspx?action=spf%3astuffins.dev
[2] https://mxtoolbox.com/SuperTool.aspx?action=dkim%3astuffins.dev%3apostal-SsNdPY

**Comment by willpower232** [comment]
You can try running nslookup or dig on the server itself, you may need to make use of the setting `use_local_ns_for_domains` https://github.com/postalserver/postal/blob/d808d15c28d1c4c626af44bedc05743b01e078b9/config/postal.defaults.yml#L17

**Reply by paulstuffins** [reply]
Hi Will,

The domain tld I was testing with, a .dev, will not return any values, not even SOA or NS, if the domain does not have DNSSEC set up for it, unless you explicitly tell it to ignore DNSSEC failures.

I have set it up with a .co.uk domain, and it picked the SPF and DKIM records up immediately.

But thanks for your help!

**Reply by willpower232** [reply]
Interesting, I hadn't heard of this being a thing. I wonder if there is some setting more local to your provider enforcing these conditions.
